### PR TITLE
add get for unsigned ints from univalue class

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

**Comment by sickpig** [comment]
Didn't look at the changes but didn't we manage the univalue code as a subtree of the upstream repo?

**Comment by Greg-Griffith** [comment]
If it passes review here. i accept the task of making a PR to merge the changes into the upstream repo. There is no way to get an unsigned integer of any size out of univalue yet it accepts them as constructors. This seems inconsistent to me and this PR addresses that

File: src/univalue/lib/univalue_get.cpp [hunk 1]
```diff
@@ -58,6 +58,22 @@ bool ParseInt64(const std::string& str, int64_t *out)
         n <= std::numeric_limits<int64_t>::max();
 }
 
+
+bool ParseUint64(const std::string& str, uint64_t *out)
+{
+    if (!ParsePrechecks(str))
+        return false;
+    char *endp = NULL;
```
Comment (sickpig) [inline src/univalue/lib/univalue_get.cpp:66]: nullptr here

**Review by sickpig — APPROVED** [review]
utack. 
hopefully it will be accepted upstream

**Comment by ptschip** [comment]
@Greg-Griffith create a .diff and add it the  /patches folder...you'll see a couple of other patches there as examples.  IMO the best would be to get it merged upstream first (if possible) and then pull in the repo after.  If you look at the patches folder you'll see not many, the only recent one from BU was the leveldb but that was a config change and so wasn't something that would get merged upstream.

File: src/univalue/lib/univalue_get.cpp (hunk 1 above)
Comment (Greg-Griffith) [reply src/univalue/lib/univalue_get.cpp:66]: ill update all of the parse functions for this

**Comment by Greg-Griffith** [comment]
@ptschip i did not know that. I will do that.

**Comment by Greg-Griffith** [comment]
i will also add a .diff for #1589 in a different PR

**Comment by gandrewstone** [comment]
unit test?

**Comment by Greg-Griffith** [comment]
cherry-picked and rebased so these changes match the changes i submitted to upstream here: https://github.com/jgarzik/univalue/pull/60

**Pull request merged**